pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -72,22 +72,33 @@ def __init__(
     ) -> None:
         super().__init__(parallel_devices=parallel_devices, cluster_environment=cluster_environment)
         self.interactive_ddp_procs = []
-        self.num_nodes = num_nodes
+        self._num_nodes = num_nodes
         self.sync_batchnorm = sync_batchnorm
         self.dist = LightningDistributed()
+        self.num_processes = len(self.parallel_devices) if self.parallel_devices is not None else 0
         self._ddp_kwargs = kwargs
         self._has_spawned_children = False
         self.task_idx = None
-        self.num_processes = len(parallel_devices) if parallel_devices is not None else parallel_devices
         self._ddp_comm_state = ddp_comm_state
         self._ddp_comm_hook = ddp_comm_hook
         self._ddp_comm_wrapper = ddp_comm_wrapper
+        # note that world ranks is related to num_nodes, when resetting these parameters,
+        # need to reset world ranks
```

Comment:
I believe this comment is better in the setter below :)

Reply:
@awaelchli imo we should also mark these as arguments as deprecated in the constructor. Wdyt?